Formula/glibc.rb

```diff
@@ -95,41 +95,160 @@ class Glibc < Formula
 
   depends_on "binutils" => :build
   depends_on GawkRequirement => :build
-  depends_on "linux-headers@4.4" => :build
+  depends_on "linux-headers@5.15" => :build
   depends_on MakeRequirement => :build
   depends_on SedRequirement => :build
   depends_on BrewedGlibcNotOlderRequirement
   depends_on :linux
   depends_on LinuxKernelRequirement
 
-  # GCC 4.7 or later is required.
-  fails_with gcc: "4.6"
+  # Vendor build dependencies that are needed because the host versions
+  # may be unavailable or too old.
+  resource "make" do
+    url "https://ftp.gnu.org/gnu/make/make-4.3.tar.gz"
```

Comment:
How/when will these get updated?

Reply:
I updated the comment here so that we know to check the INSTALL file to see if anything needs to be updated.

Reply:
@danielnachun I was also meaning "how would we know if these have security vulnerabilities we need to address"?

Reply:
That's a fair question but I'm not sure that has to be a major concern here.  These are build time-only dependencies so they won't be around once the build is done and won't even be shipped to users if they can use the bottle.  I suppose the only hypothetical concern would be if one of these tools was found to create a vulnerability in the compiled `glibc` code.  But given how foundational these tools are and how incredibly conservative and heavily audited `glibc` is, this seems like a far-fetched concern at the moment.

Reply:
> I suppose the only hypothetical concern would be if one of these tools was found to create a vulnerability in the compiled `glibc` code.

Yeh, that's my concern really. 

Out of interest: how were the current versions chosen and how would updates be chosen?

Just for brain-dumping rather than anything actionable: it occurred to me that this formula feels a lot like `portable-ruby` and friends in a bunch of ways; I almost wonder if building it in the same way would be useful 🤔 (particularly if we also were ever considering `portable-curl` or `portable-git`). CC @bo98 in case you have thoughts here.

Reply:
Yeah all of this seems quite heavy and increases the surface area for failure.

The main problem with a "portable-glibc" is actually making Glibc... portable (relocatable). It doesn't support that on its own so will mean binary patching.

If it is at all possible to have a bottle we can relocate, I'd really like to pursue doing so as it would make the build-from-source considerations less of an issue. It might be tricky though - I don't know.

Reply:
> If it is at all possible to have a bottle we can relocate, I'd really like to pursue doing so as it would make the build-from-source considerations less of an issue. It might be tricky though - I don't know.

Agreed. The build-from-source story for this `glibc` formula seems pretty horrific with all the vendoring 😬.

Reply:
Getting to a relocatable bottle is the goal!  It will just take a bit more work that isn't there yet.  I will say that although we are doing a lot of vendoring, all of the dependencies are really, really simple.  Python has a lot of optional build dependencies but it will just skip anything that is missing, so it ends up being super easy.  GCC also seems a bit complicated but really it's just building a barebones compiler and the arguments are just to disabled stuff we don't need.

So while I do not love doing it this way, I am quite confident this will actually be very robust.  I've tested it in many different configurations and it has worked very well.  We always consider non-default prefixes to have "best effort" support and I consider this my "best effort" at the moment.  I'm hoping we can be flexible here because of the special status of `glibc` and see how this works.  If we get hit with a lot of weird problem reports I'll have to revisit what I've done here.  

My hope here is just to make this a seamless experience for users so that we get as little as friction as possible during the migration.

Reply:
> I will say that although we are doing a lot of vendoring, all of the dependencies are really, really simple.

What is the build time pre and post vendoring?

> If we get hit with a lot of weird problem reports I'll have to revisit what I've done here.

I'm a bit concerned by the uncertainty here.